docs/adr/0006-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,122 @@
+  * Firefox iOS: ?
+  * Firefox Desktop: ?
+
+* Date: Dec 19, 2022
+* Feedback deadline: Jan 9, 2022
+
+## Context and Problem Statement
+
+Application services currently has a long cycle time for code changes.  We make a release from main on an ad-hoc schedule, averaging about 3-4 releases a month.  Our consumer applications take our releases also on an ad-hoc basis, `firefox-android` takes our releases fairly frequently, `firefox-ios` less frequently, and desktop very infrequently.  This means that code changes take a relatively long time to be released to the public.  For example when the `v96.0.1` release was taken by `firefox-android`, the new commits for that release ranged from 3-18 days old.
+
+There are two main reasons for our long cycles:
+  - Our current releases process requires a good deal of developer involvement.  Normal releases take a couple hours, during which a developer needs to monitor the progress and perform several manual steps along the way.  If the unreleased code has breaking changes, then the developer also needs to merge the PR for the consumer application(s) after the release, which is not ideal since that developer is often not the one who wrote the PR.  Worse, if the dev who caused the breaking changes forgot to create a PR, they need to create that PR now, leading to multiple devs coordinating for the release.  Releases that are not from main also require [extra work](https://github.com/mozilla/application-services/blob/6ef3a2f2c4fef822a539e0fa375683c6acfb2617/docs/howtos/cut-a-new-release.md#make-a-new-point-release-from-an-existing-release-that-is-behind-latest-main): creating a branch for that version, backporting the fix, making a patch release, etc.
+  - Both application-services team and the application teams generally put low priority on updating the application-services version.  We generally focus on the benefits of waiting to take releases rather than the costs.
+
+Mozilla generally favors a rapid release model to long cycle times.  Firefox desktop adopted rapid releases over a decade ago and there is current work to move both the mobile applications and our services towards rapid releases.  Although there are some costs to rapid releases, as a company we've decided that we're willing to accept them.
+
+Application services should consider moving towards a rapid release model.  The same reasoning that led other teams to adopt rapid releases also applies to us.  Furthermore, the mobile teams are moving towards rapid releases which may depend on us also adopting them.  Finally, we are in the process of moving towards a single monorepo for android code, with potential to add iOS code and/or move into the moz-central monorepo. This would require that application services code is taken by all other consumer applications in the monorepo as soon as it's merged.  Reducing the amount of time between our releases and when applications take those releases brings us closer to that monorepo world.
```

Comment:
One bit that confused me here. You write that Firefox Desktop adopted rapid releases over a decade ago, and mobile is moving toward it. Mobile is already on pretty much the same release cadence as Desktop, or did this specifically refer to iOS, where the cadence is the same, but Nightly|Beta is TestFlight only?

Reply:
(The ongoing rapid release project aims to increase cadence significantly relative to that...on mobile).

Reply:
Good point, I think I got confused by discussions of the "new rapid release model".  I'm going to rewrite this to say:
  - Mobile has more-or-less the same release cadence as desktop
  - There's work going on to speed it up even more.